src/clojure_lsp/main.clj

```diff
@@ -315,24 +307,18 @@
 
 (deftype LSPWorkspaceService []
   WorkspaceService
+
   (^CompletableFuture executeCommand [_ ^ExecuteCommandParams params]
-    (go :executeCommand
-        (let [[doc-id line col & args] (map interop/json->clj (.getArguments params))
-              command (.getCommand params)]
-          (future
-            (end
-              (try
-                (when-let [result (#'handlers/refactor doc-id
-                                                       (inc (int line))
-                                                       (inc (int col))
-                                                       command
-                                                       args)]
-                  (.get (.applyEdit (:client @db/db)
-                                    (ApplyWorkspaceEditParams.
-                                      (interop/conform-or-log ::interop/workspace-edit result)))))
-                (catch Exception e
-                  (log/error e)))))))
-    (CompletableFuture/completedFuture 0))
+   (go :executeCommand
```

Comment:
My reading of ExecuteCommand is that it doesn't expect results back, but rather triggers a server->client message elsewhere. So I don't think this should be switched to supplyAsync.

Rather the server-info branch would probably call `(.showMessage (:client @db/db) conformed result)`

Reply:
Actually `executeCommand` can return `any` response according to spec, but for this case, I think using a `showMessage` would suit better for what we need.